fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -342,6 +339,344 @@ public void updateDatabaseUsedQuotaData(long usedQuotaDataBytes) {
+                                    Joiner.on(",").join(errorBackendIdsForTablet));
+                            throw new TabletQuorumFailedException(transactionId, tablet.getId(),
+                                    successReplicaNum, quorumReplicaNum,
+                                    errorBackendIdsForTablet);
+                        }
+                    }
+                }
+            }
+        }
+
+        unprotectedPreCommitTransaction(transactionState, errorReplicaIds, tableToPartition, totalInvolvedBackends, db);
+        LOG.info("transaction:[{}] successfully pre-committed", transactionState);
+    }
+
+    public TransactionState checkPreCommitStatus(long transactionId) throws UserException {
+        Database db = catalog.getDbOrMetaException(dbId);
+        TransactionState transactionState;
+        readLock();
+        try {
+            transactionState = unprotectedGetTransactionState(transactionId);
+        } finally {
+            readUnlock();
+        }
+
+        if (transactionState == null) {
+            LOG.debug("transaction not found: {}", transactionId);
+            throw new TransactionCommitFailedException("transaction {" + transactionId + "} not found.");
+        }
+
+        transactionState.setCheckTimeout(false);
+
+        if (transactionState.getTransactionStatus() == TransactionStatus.ABORTED) {
+            LOG.debug("transaction is already aborted: {}", transactionId);
+            throw new TransactionCommitFailedException("transaction [" + transactionId
+                    + "] is already aborted, not pre-committed" + transactionState.getReason());
+        }
+
+        if (transactionState.getTransactionStatus() == TransactionStatus.VISIBLE) {
+            LOG.debug("transaction is already visible: {}", transactionId);
+            throw new TransactionCommitFailedException("transaction [" + transactionId
+                    + "] is already visible, not pre-committed");
+        }
+
+        if (transactionState.getTransactionStatus() == TransactionStatus.COMMITTED) {
+            LOG.debug("transaction is already committed: {}", transactionId);
+            throw new TransactionCommitFailedException("transaction [" + transactionId
+                    + "] is already committed, not pre-committed");
+        }
+
+        if (transactionState.getTransactionStatus() == TransactionStatus.PREPARE) {
+            LOG.debug("transaction is prepare, not pre-committed: {}", transactionId);
+            throw new TransactionCommitFailedException("transaction [" + transactionId
+                    + "] is prepare, not pre-committed");
+        }
+
+        long currentTimeMillis = System.currentTimeMillis();
+        // Maybe new  invisible version has not been reported to master FE after the transaction is pre-committed
+        // or FE restart, we should wait a little while to commit transaction.
+        if (currentTimeMillis - transactionState.getPreCommitTime() <
```

Comment:
Does it mean that we need to wait at least 60sec to commit the precommitted the txn?
That is strange to me.